File: app/frontend/ea-project/src/models/output/GetAllClubsOutput.js
import BaseOutput from "./BaseOutput";

export default class GetAllClubsOutput extends BaseOutput {
  constructor(clubList = [], feedbackMessages = []) {
    super(feedbackMessages);
    this.clubList = clubList.map(item => new GetAllClubsOutput.ClubProperties(item));
  }

  static ClubProperties = class {
    constructor({id = 0, name = '', isActive = false, creationDate = new Date()}) {
      this.id = id;
      this.name = name;
      this.isActive = isActive;
      this.creationDate = creationDate;
    }
  };
}

File: app/backend/src/main/java/eaproject/output/GetAllClubsOutput.java
package eaproject.output;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.io.Serializable;
import java.sql.Timestamp;
import java.util.ArrayList;

public class GetAllClubsOutput extends BaseOutput implements Serializable {

    @JsonProperty
    private ArrayList<ClubProperties> clubList;

    public ArrayList<ClubProperties> getClubList() {
        return clubList;
    }

    public void setClubList(ArrayList<ClubProperties> clubList) {
        this.clubList = clubList;
    }

    public static class ClubProperties implements Serializable {

        @JsonProperty
        private Integer id;

        @JsonProperty
        private String name;

        @JsonProperty
        private boolean isActive;

        @JsonProperty
        private Timestamp creationDate;

        public Integer getId() {
            return id;
        }

        public void setId(Integer id) {
            this.id = id;
        }

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }

        public boolean getIsActive() {
            return isActive;
        }

        public void setIsActive(boolean isActive) {
            this.isActive = isActive;
        }

        public Timestamp getCreationDate() {
            return creationDate;
        }

        public void setCreationDate(Timestamp creationDate) {
            this.creationDate = creationDate;
        }
    }
}
